src/lib/registration/widgets/registration_code.rb

```diff
@@ -73,13 +74,38 @@ def register
         Storage::InstallationOptions.instance.base_registered = true
       end
 
+      def validate
+        return error(_("System already registered.")) if Registration.is_registered?
+
+        return true if skip?
+
+        return error(_("Not valid url.")) if url? & !valid_url?
```

Comment:
I suggest to write some help here, like "Only supported schemas are http and https"?